modules/dnn/test/test_layers.cpp

```diff
@@ -618,6 +618,1731 @@ TEST(Layer_LSTM_Test_Accuracy_with_, HiddenParams)
+    std::vector<int> output_shape = {1, 3};
+    if (batch_size == 0) {
+        input_shape.erase(input_shape.begin());
+        output_shape.erase(output_shape.begin());
+    }
+
+    float inp_value = 1 + static_cast <float> (rand()) / (static_cast <float> (RAND_MAX/(10-1)));
+    cv::Mat input = cv::Mat(input_shape, CV_32F, inp_value);
+    cv::Mat output_ref = cv::Mat(output_shape, CV_32F, 1.0);
+    std::vector<Mat> inputs{input};
+    std::vector<Mat> outputs;
+
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    normAssert(output_ref, outputs[0]);
+}
+INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Clip_1d_Test, Values(0, 1));
+
+typedef testing::TestWithParam<tuple<int>> Layer_ReLU_1d_Test;
+TEST_P(Layer_ReLU_1d_Test, Accuracy)
+{
+    int batch_size = get<0>(GetParam());
+    LayerParams lp;
+
+    lp.type = "ReLU";
+    lp.name = "reluLayer";
+    lp.set("negative_slope", 0.0);
+    Ptr<ReLULayer> layer = ReLULayer::create(lp);
+
+    std::vector<int> input_shape = {1, 3};
+    std::vector<int> output_shape = {1, 3};
+    if (batch_size == 0) {
+        input_shape.erase(input_shape.begin());
+        output_shape.erase(output_shape.begin());
+    }
+
+    float inp_value = 1 + static_cast <float> (rand()) / (static_cast <float> (RAND_MAX/(10-1)));
+    cv::Mat input = cv::Mat(input_shape, CV_32F, inp_value);
+    cv::Mat output_ref = cv::Mat(output_shape, CV_32F, inp_value);
+    std::vector<Mat> inputs{input};
+    std::vector<Mat> outputs;
+
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    normAssert(output_ref, outputs[0]);
+}
+INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_ReLU_1d_Test, Values(0, 1));
+
+typedef testing::TestWithParam<tuple<int>> Layer_GeLU_1d_Test;
+TEST_P(Layer_GeLU_1d_Test, Accuracy)
+{
+    int batch_size = get<0>(GetParam());
+    LayerParams lp;
+
+    lp.type = "GeLU";
+    lp.name = "geluLayer";
+    Ptr<GeluLayer> layer = GeluLayer::create(lp);
+
+    std::vector<int> input_shape = {1, 3};
+    std::vector<int> output_shape = {1, 3};
+    if (batch_size == 0) {
+        input_shape.erase(input_shape.begin());
+        output_shape.erase(output_shape.begin());
+    }
+
+    float inp_value = 1 + static_cast <float> (rand()) / (static_cast <float> (RAND_MAX/(10-1)));
+    cv::Mat input = cv::Mat(input_shape, CV_32F, inp_value);
+    // GeLU computation
+    float geluValue = inp_value * 0.5 * (1.0 + std::tanh(std::sqrt(2.0 / M_PI) * (inp_value + 0.044715 * std::pow(inp_value, 3))));
+    float data[3] = {geluValue, geluValue, geluValue};
+    cv::Mat output_ref = cv::Mat(output_shape, CV_32F, data);
+    std::vector<Mat> inputs{input};
+    std::vector<Mat> outputs;
+
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    // Note: Adjust the tolerance as needed based on empirical testing
+    normAssert(output_ref, outputs[0], "", 1e-1, 1e-1);
```

Comment:
0.1 looks like too weak threshold.

Reply:
Otherwise results diverge. Not sure why.

Reply:
@fengyuentau could you take a look? Looks like some implementation bug.

Reply:
Gelu:

https://github.com/opencv/opencv/blob/5e5a035c5bddbd0190f6cec8e9db90363f402799/modules/dnn/src/layers/elementwise_layers.cpp#L827-L830

GeluApproximation:

https://github.com/opencv/opencv/blob/5e5a035c5bddbd0190f6cec8e9db90363f402799/modules/dnn/src/layers/elementwise_layers.cpp#L862-L866

@Abdurrahheem You are calculating approximated Gelu and comapring with Gelu.